algorithms/src/polycommit/sonic_pc/mod.rs

```diff
@@ -767,221 +605,54 @@ impl<E: PairingEngine> PolynomialCommitment<E::Fr, E::Fq> for SonicKZG10<E> {
-
-        Self::accumulate_elems_individual_opening_challenges(
-            &mut combined_comms,
-            &mut combined_witness,
-            &mut combined_adjusted_witness,
-            vk,
-            commitments,
-            *point,
-            values,
-            proof,
-            opening_challenges,
-            None,
-        );
-
-        let res = Self::check_elems(combined_comms, combined_witness, combined_adjusted_witness, vk);
-        end_timer!(check_time);
-        res
+        combined_comm
     }
 
-    /// batch_check but with individual challenges
-    fn batch_check_individual_opening_challenges<'a, R: RngCore>(
-        vk: &<Self as PolynomialCommitment<E::Fr, E::Fq>>::VerifierKey,
-        commitments: impl IntoIterator<
-            Item = &'a LabeledCommitment<<Self as PolynomialCommitment<E::Fr, E::Fq>>::Commitment>,
-        >,
-        query_set: &QuerySet<E::Fr>,
-        evaluations: &Evaluations<E::Fr>,
-        proof: &<Self as PolynomialCommitment<E::Fr, E::Fq>>::BatchProof,
-        opening_challenges: &dyn Fn(u64) -> E::Fr,
-        _rng: &mut R,
-    ) -> Result<bool, PCError>
-    where
-        <Self as PolynomialCommitment<E::Fr, E::Fq>>::Commitment: 'a,
-    {
-        let commitments: BTreeMap<_, _> = commitments.into_iter().map(|c| (c.label(), c)).collect();
-
-        let mut query_to_labels_map = BTreeMap::new();
-        for (label, (point_name, point)) in query_set.iter() {
-            let labels = query_to_labels_map.entry(point_name).or_insert((point, BTreeSet::new()));
-            labels.1.insert(label);
-        }
-
-        // Implicit assumption: proofs are order in same manner as queries in
-        // `query_to_labels_map`.
-        let proofs: Vec<_> = proof.clone();
-        assert_eq!(proofs.len(), query_to_labels_map.len());
-
-        let mut result = true;
-        for ((_point_name, (point, labels)), proof) in query_to_labels_map.into_iter().zip_eq(proofs) {
-            let mut comms: Vec<&'_ LabeledCommitment<_>> = Vec::new();
-            let mut values = Vec::new();
-            for label in labels {
-                let commitment =
-                    commitments.get(label).ok_or(PCError::MissingPolynomial { label: label.to_string() })?;
-
-                let v_i = evaluations
-                    .get(&(label.clone(), *point))
-                    .ok_or(PCError::MissingEvaluation { label: label.to_string() })?;
-
-                comms.push(commitment);
-                values.push(*v_i);
-            }
-
-            let proof_time = start_timer!(|| "Checking per-query proof");
-            result &= Self::check_individual_opening_challenges(vk, comms, point, values, &proof, opening_challenges)?;
-            end_timer!(proof_time);
+    fn normalize_commitments(commitments: Vec<E::G1Projective>) -> impl Iterator<Item = Commitment<E>> {
+        let mut comms = Vec::with_capacity(commitments.len());
```

Comment:
doesn't this vector contain the same contents as `commitments`? if I'm not missing anything, `comms` can be removed and `commitments` used directly

Reply:
Oh good point.